265 boss

l3nny123 opened the pull request:
Added a boss (blue giant):
- spawns in wave 2, the normal wave will also spawn (normal wave 2 + Boss) 
- the boss has a melee attack 
-  There is a 1/2 probability that a range attack will be executed with every melee attack or if you enter a certain area around the boss.
- range attack spawns 5 rocks in different directions which deal damage

Pazl27 commented:
i tried to run it an revised theses error in waves two: 
```
  ERROR: HEADLESS STDERR: Unknown ScenePath: res://Entities/Enemies/GiantBoss/GiantBossEnemy.tscn
  ERROR: HEADLESS STDERR: Unknown ScenePath: res://Entities/Enemies/GiantBoss/GiantBossEnemy.tscn
  ERROR: HEADLESS STDERR: Unknown ScenePath: res://Entities/Enemies/GiantBoss/GiantBossEnemy.tscn
  ERROR: HEADLESS STDERR: Unknown ScenePath: res://Entities/Enemies/GiantBoss/GiantBossEnemy.tscn
  ERROR: HEADLESS STDERR: Unknown ScenePath: res://Entities/Enemies/GiantBoss/GiantBossEnemy.tscn

```

l3nny123 commented:
they dont show up for me idk

l3nny123 commented:
I have just seen that the file name in your error message is wrong. The G should be small “giantBossEnemy.tscn”. I also had this error during development because I had renamed the file. I have seen that the name of a node was wrong. I hope it works now.

manuel0804 commented:
I still get the same error

manuel0804 commented:
Ah nevermind it works, my godot is just stupid dumb

manuel0804 reviewed (changes requested):
Really cool and really nice assets!
But maybe one small improvement: The boss is so slow in general and his attack speed is so low. I neither saw any meele attack nor the rock attack. IF you want you can scale this a bit. However its really nice in general.

l3nny123 commented:
the boss should be faster now. if you stand right next to the boss for a moment you should be able to see the melee attack animation an the rocks. (you can also try to walk away from the boss so you dont see him anymore if you come back there is a 1/2 chance again for him to do the rock attack)

l3nny123 commented:
idk why. I mean the file was namend "GiantBossEnemy.tscn" at one point but i renamed it and changed every reference i had set manually to the new name "giantBossEnemy.tscn"

l3nny123 commented:
i changed something maybe it works now. The problem is I dont get the error so i cant really check if it works
